Added best sorting to frontpage

ThatsEli opened the pull request:
I added the missing mixin for best sorting. This is only available on the fontpage, but is the default sorting type and should be available. There was already an issue(#143), but it was closed due to only being needed for the fontpage. I added it as a separate mixin so it can be only applied to the fontpage. I hope this is the right approach to doing this, if not a hint at the right way would be nice and I would try to implement it that way. Due to me being new to Flutter/Dart, there can be some beginner mistakes.  I hope the test are okay as they are. 👍

bkonyi commented on lib/src/listing/mixins/best.dart:
```diff
@@ -0,0 +1,26 @@
+  /// (the default is 100). If provided, `after` specifies from which point
+  /// Reddit will return objects of the requested type. `params` is a set of
+  /// additional parameters that will be forwarded along with the request.
+  Stream<UserContent> best(
```
Since this is only used by `FrontPage` we don't need a mixin. I'd just move `best(...)` into `FrontPage`.

bkonyi commented on lib/src/listing/mixins/best.dart:
```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2017, the Dart Reddit API Wrapper project authors.
```
General comment: when adding new files, the copyright year should be the current year.

bkonyi commented on test/best_listing_mixin/best_listing_mixin_test.dart:
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2018, the Dart Reddit API Wrapper project authors.
```
General comment: when adding new files, the copyright year should be the current year.

bkonyi commented on test/best_listing_mixin/best_listing_mixin_test.dart:
```diff
@@ -0,0 +1,21 @@
+import '../test_utils.dart';
+
+Future<void> main() async {
+  test('lib/best_listing_mixin/best_sanity', () async {
```
Thanks for writing a simple test!

bkonyi commented on lib/src/frontpage.dart:
```diff
@@ -7,12 +7,17 @@ import 'package:draw/src/base.dart';
+        BaseListingMixin,
+        GildedListingMixin,
+        RisingListingMixin,
+        BestListingMixin {
```
General comment: sort mixins by name.

bkonyi commented on lib/src/frontpage.dart:
```diff
@@ -7,12 +7,17 @@ import 'package:draw/src/base.dart';
 import 'package:draw/src/listing/mixins/base.dart';
 import 'package:draw/src/listing/mixins/gilded.dart';
 import 'package:draw/src/listing/mixins/rising.dart';
+import 'package:draw/src/listing/mixins/best.dart';
```
General comment: sort imports in alphabetical order.

bkonyi commented on test/test_all.dart:
```diff
@@ -15,6 +15,8 @@ import 'multireddit/multireddit_test.dart' as multireddit_test;
 import 'redditor/redditor_test.dart' as redditor_test;
 import 'rising_listing_mixin/rising_listing_mixin_test.dart'
     as rising_listing_mixin_test;
+import 'best_listing_mixin/best_listing_mixin_test.dart'
```
General comment: sort imports in alphabetical order.

bkonyi commented on test/test_all.dart:
```diff
@@ -41,6 +43,7 @@ void main() {
   multireddit_test.main();
   redditor_test.main();
   rising_listing_mixin_test.main();
+  best_listing_mixin_test.main();
```
General comment: sort these by alphabetical order (`live_auth_test` is sort of an exception).

ThatsEli commented:
Firstly, thank you very much for your tips :) I moved the best(...) function into the frontpage and fixed the other issues. The tests are also adjusted to this. And I only did some basic copy/paste, but thank you!
